Cubical/HITs/AssocList/Properties.agda

```diff
@@ -72,6 +76,108 @@ FMS→AL∘AL→FMS≡id = AL.ElimProp.f (AL.trunc _ _) refl (λ x n {xs} p →
+ALmember-per*-aux a x y xs (yes a≡x) (no  a≢y) = refl
+ALmember-per*-aux a x y xs (no  a≢x) (yes a≡y) = refl
+ALmember-per*-aux a x y xs (no  a≢x) (no  a≢y) = refl
+
+
+ALmember-agg*-aux :  (a x : A) (m n xs : ℕ) (p : Dec (a ≡ x))
+                   →  ALmember-⟨,⟩∷*-aux a x p m (ALmember-⟨,⟩∷*-aux a x p n xs)
+                    ≡ ALmember-⟨,⟩∷*-aux a x p (m + n) xs
+ALmember-agg*-aux a x m n xs (yes x≡a) = +-assoc m n xs
+ALmember-agg*-aux a x m n xs (no  x≢a) = refl
+
+
+ALmember-del*-aux :  (a x : A) (xs : ℕ) (p : Dec (a ≡ x))
+                   →  ALmember-⟨,⟩∷*-aux a x p 0 xs ≡ xs
+ALmember-del*-aux a x xs (yes a≡x) = refl
+ALmember-del*-aux a x xs (no  a≢x) = refl
+
+
+
+module _(discA : Discrete A) where
+ ALmember-⟨,⟩∷* : A → A → ℕ → ℕ → ℕ
+ ALmember-⟨,⟩∷* a x n xs = ALmember-⟨,⟩∷*-aux a x (discA a x) n xs
+
+
+ ALmember-per* :  (a x y : A) (xs : ℕ)
+                →  ALmember-⟨,⟩∷* a x 1 (ALmember-⟨,⟩∷* a y 1 xs)
+                 ≡ ALmember-⟨,⟩∷* a y 1 (ALmember-⟨,⟩∷* a x 1 xs)
+ ALmember-per* a x y xs = ALmember-per*-aux a x y xs (discA a x) (discA a y)
+
+
+ ALmember-agg* :  (a x : A) (m n xs : ℕ)
+                →  ALmember-⟨,⟩∷* a x m (ALmember-⟨,⟩∷* a x n xs)
+                 ≡ ALmember-⟨,⟩∷* a x (m + n) xs
+ ALmember-agg* a x m n xs = ALmember-agg*-aux a x m n xs (discA a x)
+
+
+ ALmember-del* :  (a x : A) (xs : ℕ) → ALmember-⟨,⟩∷* a x 0 xs ≡ xs
+ ALmember-del* a x xs = ALmember-del*-aux a x xs (discA a x)
+
+
+ ALmember : A → AssocList A → ℕ
+ ALmember a = AL.Rec.f isSetℕ 0 (ALmember-⟨,⟩∷* a) (ALmember-per* a) (ALmember-agg* a) (ALmember-del* a)
+
+
+
+ AL-with-str : Multi-Set A (Discrete→isSet discA)
+ AL-with-str = (AssocList A , ⟨⟩ , ⟨_, 1 ⟩∷_ , ALmember)
+
+
+-- We want to show that Al-with-str ≅ FMS-with-str as multiset-structures
+module _(discA : Discrete A) where
+ FMS→AL-isIso : multi-set-iso A (Discrete→isSet discA) (FMS-with-str discA) (AL-with-str discA) FMSet≃AssocList
+ FMS→AL-isIso = refl , (λ a xs → refl) , φ
+  where
+  φ : ∀ a xs → FMSmember discA a xs ≡ ALmember discA a (FMS→AL xs)
+  φ a = FMS.ElimProp.f (isSetℕ _ _) refl ψ
+   where
+   χ :  (x : A) (xs ys : ℕ) (p : Dec (a ≡ x))
+      → xs ≡ ys
+      → FMSmember-∷*-aux a x p xs ≡ ALmember-⟨,⟩∷*-aux a x p 1 ys
+   χ x xs ys (yes p) q = cong suc q
+   χ x xs ys (no ¬p) q = q
+
+   ψ :  (x : A) {xs : FMSet A}
+      → FMSmember discA a xs ≡ ALmember discA a (FMS→AL xs)
+      → FMSmember discA a (x ∷ xs) ≡ ALmember discA a (FMS→AL (x ∷ xs))
+   ψ x {xs} p = subst B α θ
+    where
+    B = λ ys → FMSmember discA a (x ∷ xs) ≡ ALmember discA a ys
+
+    α : ⟨ x , 1 ⟩∷ FMS→AL xs ≡ FMS→AL (x ∷ xs)
+    α = sym (multi-∷-id x 1 xs)
+
+    θ : FMSmember discA a (x ∷ xs) ≡ ALmember discA a (⟨ x , 1 ⟩∷ (FMS→AL xs))
+    θ = χ x (FMSmember discA a xs) (ALmember discA a (FMS→AL xs)) (discA a x) p
+
+
+
+ FMS-with-str≡AL-with-str : FMS-with-str discA ≡ AL-with-str discA
+ FMS-with-str≡AL-with-str = SIP (multi-set-structure A (Discrete→isSet discA))
```

Comment:
Instead of calling `Discrete→isSet discA` repeatedly just make it a private definition in the model.

Reply:
module :-)